packages/admin-panel/src/routes/visualisations/mapOverlayGroupRelations.js

```diff
@@ -7,64 +7,92 @@ const RESOURCE_NAME = { singular: 'map overlay group relation' };
 
 export const RELATION_ENDPOINT = 'mapOverlayGroupRelations';
 
-const FIELDS = [
-  {
-    Header: 'Map overlay group code',
-    source: 'map_overlay_group.code',
-
-    editConfig: {
-      optionsEndpoint: 'mapOverlayGroups',
-      optionLabelKey: 'map_overlay_group.code',
-      optionValueKey: 'map_overlay_group.id',
-      sourceKey: 'map_overlay_group_id',
-    },
+const mapOverlayGroupCode = {
+  Header: 'Map overlay group code',
+  source: 'map_overlay_group.code',
+  editConfig: {
+    optionsEndpoint: 'mapOverlayGroups',
+    optionLabelKey: 'map_overlay_group.code',
+    optionValueKey: 'map_overlay_group.id',
+    sourceKey: 'map_overlay_group_id',
   },
-  {
-    Header: 'Child ID',
-    source: 'child_id',
+};
 
-    editConfig: {
-      optionsEndpoint: 'mapOverlays',
-      optionLabelKey: 'mapOverlay.id',
-      optionValueKey: 'mapOverlay.id',
-      canCreateNewOptions: true,
-      sourceKey: 'child_id',
-    },
+const childType = {
+  Header: 'Child type',
+  width: 160,
+  source: 'child_type',
+  editConfig: {
+    options: [
+      {
+        label: 'Map overlay',
+        value: 'mapOverlay',
+      },
+      {
+        label: 'Map overlay group',
+        value: 'mapOverlayGroup',
+      },
+    ],
   },
-  {
-    Header: 'Child type',
-    width: 160,
-    source: 'child_type',
+};
 
-    editConfig: {
-      options: [
-        {
-          label: 'Map overlay',
-          value: 'mapOverlay',
-        },
-        {
-          label: 'Map overlay group',
-          value: 'mapOverlayGroup',
-        },
-      ],
-    },
+const childCode = {
+  Header: 'Child Code',
+  source: 'child_code',
+};
+
+const childMapOverlayCode = {
+  Header: 'Child map overlay code',
+  id: 'child_map_overlay_code',
+  source: 'child_code',
+  editConfig: {
+    optionsEndpoint: 'mapOverlays',
+    optionLabelKey: 'mapOverlay.code',
+    optionValueKey: 'mapOverlay.id',
+    sourceKey: 'child_id',
+    visibilityCriteria: { child_type: 'mapOverlay' },
   },
-  {
-    Header: 'Sort order',
-    source: 'sort_order',
+};
+
+const childMapOverlayGroupCode = {
+  Header: 'Child map overlay group code',
+  id: 'child_map_overlay_group_code',
+  source: 'child_code',
+  editConfig: {
+    optionsEndpoint: 'mapOverlayGroups',
+    optionLabelKey: 'mapOverlayGroups.code',
+    optionValueKey: 'mapOverlayGroups.id',
+    sourceKey: 'child_id',
+    visibilityCriteria: { child_type: 'mapOverlayGroup' },
   },
+};
+
+const sortOrder = {
+  Header: 'Sort order',
+  source: 'sort_order',
+};
+
+const EDIT_FIELDS = [
+  mapOverlayGroupCode,
+  childType,
+  childMapOverlayCode,
+  childMapOverlayGroupCode,
+  sortOrder,
 ];
```

Comment:
This pattern doesn’t really exist in Admin Panel (we just concatenate arrays in most places), but this way order of fields always “makes sense” (e.g. **Sort order** field is always last)